src/url.rs

```diff
@@ -0,0 +1,387 @@
+                                    fdk_id, e
+                                )),
+                            }
+                        }
+                        Err(e) => Err(format!(
+                            "{} - Failed creating new metrics store: {}",
+                            fdk_id, e
+                        )),
+                    }
+                }
+                None => Err(format!("{} - Dataset node not found in graph", fdk_id)),
+            }
+        }
+        Err(e) => Err(format!("{} - Failed to parse graph: {}", fdk_id, e)),
+    }
+}
+
+fn check_urls(fdk_id: String, dataset_node: NamedNodeRef, store: &Store) -> Result<Store, Error> {
+    // Make MQA metrics model (DQV)
+    let metrics_store = create_metrics_store(dataset_node)?;
+    let distributions =
+        list_distributions(dataset_node, store).collect::<Result<Vec<Quad>, _>>()?;
+
+    for dist in distributions {
+        metrics_store.insert(
+            Quad::new(
+                dist.subject.clone(),
+                dist.predicate.clone(),
+                dist.object.clone(),
+                GraphName::DefaultGraph,
+            )
+            .as_ref(),
+        )?;
+
+        let dist_node_option = match dist.object {
+            Term::NamedNode(n) => Some(NamedOrBlankNode::NamedNode(n)),
+            Term::BlankNode(n) => Some(NamedOrBlankNode::BlankNode(n)),
+            _ => None,
+        };
+
+        match dist_node_option {
+            Some(dist_node) => {
+                info!("{} - Extracting urls from distribution", fdk_id);
+                match extract_urls_from_distribution(dist_node.as_ref(), &store) {
+                    Ok(urls) => {
+                        info!("{} - Number of urls found {}", fdk_id, urls.len());
+
+                        for url in urls {
+                            let result = check_url(url.method, url.url, url.url_type);
+
+                            info!("{}", result.note);
+
+                            let url_node: NamedNodeRef<'_> =
+                                NamedNodeRef::new_unchecked(result.url.as_str());
+
+                            match result.url_type {
+                                UrlType::AccessUrl => add_url_status_metric(
+                                    dist_node.as_ref(),
+                                    url_node,
+                                    UrlType::AccessUrl,
+                                    result.status,
+                                    &metrics_store,
+                                )?,
+                                UrlType::DownloadUrl => add_url_status_metric(
+                                    dist_node.as_ref(),
+                                    url_node,
+                                    UrlType::DownloadUrl,
+                                    result.status,
+                                    &metrics_store,
+                                )?,
+                            };
```

Comment:
Trenger ikke sjekke hva typen er her, det gjøres i funksjonen (basert på type parameteren).
```suggestion
                            add_url_status_metric(
                                dist_node.as_ref(),
                                url_node,
                                result.url_type,
                                result.status,
                                &metrics_store,
                            )?;
```